webhooks/pod_webhook.go

```diff
@@ -149,12 +151,25 @@ func (m *PodMutator) injectSidecar(pod *corev1.Pod, configMap string) ([]byte, e
 			},
 		},
 	})
+
+	commandSequence := []string{
+		"start", "--uri", "/etc/flagd/config.json",
+	}
+	// FlagD is the default provider name externally
+	if featureFlag.Spec.ServiceProvider != nil && featureFlag.Spec.ServiceProvider.Name != "FlagD" {
+		commandSequence = append(commandSequence, "--service-provider")
+		commandSequence = append(commandSequence, "http")
+	}
+	// Adds the sync provider if it is set
+	if featureFlag.Spec.SyncProvider != nil && featureFlag.Spec.SyncProvider.Name != "" {
+		commandSequence = append(commandSequence, "--sync-provider")
+		commandSequence = append(commandSequence, featureFlag.Spec.SyncProvider.Name)
```

Comment:
This may be completely paranoid, but do we need to worry about command injection at all here? Since we're appending input to a command?

Reply:
Thinking about it more, this is validated by the enum you specified above, so this shouldn't be a problem.